src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int GROWTH_FACTOR = 2;
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] elements = (T[]) new Object[DEFAULT_CAPACITY];
```

Comment:
checklist issue, this should be done in constructor

Reply:
do you mean that elements nust be passed to the inner class?